### PR TITLE
Upgrade `zeromq` to 4.3.5, `pyzmq` to 25.1.1, patch out broken tests in setup

### LINKED ISSUE #36027: {fedora-{38,39},ubuntu-mantic,debian-trixie}-minimal: pyzmq build failure
```
[pyzmq-24.0.1]     In file included from /usr/include/c++/13/ext/alloc_traits.h:34,
  [pyzmq-24.0.1]                      from /usr/include/c++/13/bits/basic_string.h:39,
  [pyzmq-24.0.1]                      from /usr/include/c++/13/string:54,
  [pyzmq-24.0.1]                      from /usr/include/c++/13/bits/locale_classes.h:40,
  [pyzmq-24.0.1]                      from /usr/include/c++/13/bits/ios_base.h:41,
  [pyzmq-24.0.1]                      from /usr/include/c++/13/ios:44,
  [pyzmq-24.0.1]                      from /usr/include/c++/13/istream:40,
  [pyzmq-24.0.1]                      from /usr/include/c++/13/sstream:40,
  [pyzmq-24.0.1]                      from bundled/zeromq/src/zmtp_engine.cpp:41:
  [pyzmq-24.0.1]     /usr/include/c++/13/bits/alloc_traits.h: In instantiation of 'struct std::__allocator_traits_base::__rebind<zmq::secure_allocator_t<unsigned char>, unsigned char, void>':
  [pyzmq-24.0.1]     /usr/include/c++/13/bits/alloc_traits.h:94:11:   required by substitution of 'template<class _Alloc, class _Up> using std::__alloc_rebind = typename std::__allocator_traits_base::__rebind<_Alloc, _Up>::type [with _Alloc = zmq::secure_allocator_t<unsigned char>; _Up = unsigned char]'
  [pyzmq-24.0.1]     /usr/include/c++/13/bits/alloc_traits.h:228:8:   required by substitution of 'template<class _Alloc> template<class _Tp> using std::allocator_traits< <template-parameter-1-1> >::rebind_alloc = std::__alloc_rebind<_Alloc, _Tp> [with _Tp = unsigned char; _Alloc = zmq::secure_allocator_t<unsigned char>]'
  [pyzmq-24.0.1]     /usr/include/c++/13/ext/alloc_traits.h:126:65:   required from 'struct __gnu_cxx::__alloc_traits<zmq::secure_allocator_t<unsigned char>, unsigned char>::rebind<unsigned char>'
  [pyzmq-24.0.1]     /usr/include/c++/13/bits/stl_vector.h:88:21:   required from 'struct std::_Vector_base<unsigned char, zmq::secure_allocator_t<unsigned char> >'
  [pyzmq-24.0.1]     /usr/include/c++/13/bits/stl_vector.h:425:11:   required from 'class std::vector<unsigned char, zmq::secure_allocator_t<unsigned char> >'
  [pyzmq-24.0.1]     bundled/zeromq/src/curve_client_tools.hpp:64:76:   required from here
  [pyzmq-24.0.1]     /usr/include/c++/13/bits/alloc_traits.h:70:31: error: static assertion failed: allocator_traits<A>::rebind_alloc<A::value_type> must be A
  [pyzmq-24.0.1]        70 |                         _Tp>::value,
  [pyzmq-24.0.1]           |                               ^~~~~
```
https://github.com/sagemath/sage/actions/runs/5706565894/job/15486087970#step:8:2141

### PR DESCRIPTION
**Pull request by mkoeppe**
Patches generated from https://github.com/zeromq/pyzmq/compare/main...mkoeppe:pyzmq:sage_fixes

- Fixes #36027
- Part of #35092

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by dimpase** [comment]
@orlitzky - can `zeromq` be updated on Gentoo? It's stuck on 4.3.4.

File: build/pkgs/pyzmq/patches/0001-setup.py-Remove-setting-of-rpath.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+                     settings['include_dirs'] += [pjoin(env, 'include')]
+                     settings['library_dirs'] += [pjoin(env, 'lib')]
+ 
+-        for path in settings['library_dirs']:
+-            _add_rpath(settings, os.path.abspath(path))
+     info(settings)
+ 
+     return settings
+-- 
+2.42.0
```
Comment (jhpalmieri) [inline build/pkgs/pyzmq/patches/0001-setup.py-Remove-setting-of-rpath.patch:24]: I'm curious: is this the `git` version number?

Comment (mkoeppe) [reply build/pkgs/pyzmq/patches/0001-setup.py-Remove-setting-of-rpath.patch:24]: Yes; `git format-patch` puts it there, I'm not sure for what exact purpose

**Comment by jhpalmieri** [comment]
LGTM

**Review by jhpalmieri — APPROVED** [review]

**Comment by mkoeppe** [comment]
Thanks!

**Comment by orlitzky** [comment]
> @orlitzky - can `zeromq` be updated on Gentoo? It's stuck on 4.3.4.

https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=dba22f41

**Pull request merged**